tests/test_service.py

```diff
@@ -10,8 +10,11 @@
 @pytest.mark.skipif(
     platform.system() == "Windows", reason="tricky to do ctrl-c in windows"
 )
-def test_service_main():
-    p = subprocess.Popen([sys.executable, "-m", "jobrunner.service"])
+def test_service_main(tmp_path):
+    p = subprocess.Popen(
+        [sys.executable, "-m", "jobrunner.service"],
+        env={"WORK_DIR": str(tmp_path), "JOB_SERVER_ENDPOINT": "https://example.com"},
+    )
```

Comment:
This is a slightly odd test in that it does't actually matter whether the requests succeed or not here. I've tried to add some comments to clarify and switched to using an unrouted IP block rather than example.com, which hangs rather than returning an error.